src/shippingservice/src/shipping_service.rs

```diff
@@ -0,0 +1,173 @@
+use opentelemetry::trace::TraceContextExt;
+use opentelemetry::trace::Tracer;
+use opentelemetry::{global, KeyValue};
+use shop::shipping_service_server::ShippingService;
+use shop::{GetQuoteRequest, GetQuoteResponse, Money, ShipOrderRequest, ShipOrderResponse};
+use tonic::{Request, Response, Status};
+
+use log::*;
+
+mod quote;
+use quote::create_quote_from_count;
+
+mod tracking;
+use tracking::create_tracking_id;
+
+const NANOS_MULTIPLE: i32 = 10000000i32;
+
+pub mod shop {
+    tonic::include_proto!("hipstershop"); // The string specified here must match the proto package name
+}
+
+#[derive(Debug, Default)]
+pub struct ShippingServer {}
+
+#[tonic::async_trait]
+impl ShippingService for ShippingServer {
+    async fn get_quote(
+        &self,
+        request: Request<GetQuoteRequest>,
+    ) -> Result<Response<GetQuoteResponse>, Status> {
+        info!("GetQuoteRequest: {:?}", request);
+
+        let itemct: u32 = request
+            .into_inner()
+            .items
+            .into_iter()
+            .fold(0, |accum, cart_item| accum + (cart_item.quantity as u32));
+
+        // We may want to ask another service for product pricing / info
+        // (although now everything is assumed to be the same price)
+        // check out the create_quote_from_count method to see how we use the span created here
+        let q = global::tracer("shippingservice/get-quote")
```

Comment:
I've tested the changes locally and I was only able to see this trace in Jaeger.
Also, the span was orphan.

Do we need to manually set the context propagation in Rust?

Reply:
Yeah we can probably extract using [`TraceContextPropagator`](https://docs.rs/opentelemetry/latest/opentelemetry/sdk/propagation/struct.TraceContextPropagator.html)

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/examples/grpc/src/server.rs#L19-L37 gives an example of how to extract propagation information from the tonic header. 

Given different grpc frameworks may store the metadata/header differently users will have to implement `Extractor` based on their grpc library.